File: objectbox-code-modifier/test-files/relation-input/Customer.java
package io.objectbox.codemodifier.test;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.annotation.Backlink;

import java.util.List;

@Entity
public class Customer {

    @Id
    long id;

    String name;

    @Backlink(to = "customerId")
    List<Order> orders;
}

File: objectbox-gradle-plugin/test-gradle-projects/kotlin-android/src/main/kotlin/io/objectbox/test/kotlin/Customer.kt
package io.objectbox.test.kotlin

import io.objectbox.annotation.Backlink
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.Index

@Entity
class Customer {

    @Id
    var id: Long = 0

    @Index
    var name: String? = null

    @Backlink(to = "customerId")
    lateinit var orders: List<Order>

    constructor()

    constructor(id: Long, name: String) {
        this.id = id
        this.name = name
    }

}
